backend/eventloop/shared_resource_loop/sharedresourceloop_test.go

```diff
@@ -0,0 +1,345 @@
+package shared_resource_loop
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/redhat-appstudio/managed-gitops/backend-shared/config/db"
+	"github.com/redhat-appstudio/managed-gitops/backend/eventloop/eventlooptypes"
+
+	sharedutil "github.com/redhat-appstudio/managed-gitops/backend-shared/util"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend/apis/managed-gitops/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Used to list down resources for deletion which are created while running tests.
+type testResources struct {
+	clusterAccess           *db.ClusterAccess
+	Clusteruser_id          string
+	Managedenvironment_id   string
+	Gitopsengineinstance_id string
+	EngineCluster_id        string
+	Clustercredentials_id   []string
+}
+
+var _ = Describe("SharedResourceEventLoop Test", func() {
+
+	// This will be used by AfterEach to clean resources
+	var resourcesToBeDeleted testResources
+
+	Context("Shared Resource Event Loop test", func() {
+
+		It("Should create or fetch a user by Namespace id.", func() {
+			// Create a fake K8s client
+			ctx := context.Background()
+
+			scheme,
+				argocdNamespace,
+				kubesystemNamespace,
+				workspace, _ := eventlooptypes.GenericTestSetup()
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-gitops-depl",
+					Namespace: workspace.Name,
+					UID:       uuid.NewUUID(),
+				},
+			}
+
+			informer := sharedutil.ListEventReceiver{}
```

Comment:
It looks like the ProxyClient/ListEventReceiver functionality isn't being used here (eg you are not reading from `Events` inside that struct for any test purposes), so you can probably just use the k8sClientOuter directly.